src/HLSMonitor.ts

```diff
@@ -309,68 +309,69 @@ export class HLSMonitor {
             }
           }
         } else {
-          // // Validate media sequence and file sequence
-          // const mseqDiff = variant.get("mediaSequence") - data.variants[bw].mediaSequence;
-          // if (mseqDiff < data.variants[bw].fileSequences.length) {
-          //   const expectedfileSequences = data.variants[bw].fileSequences[mseqDiff];
-          //   if (currentSegUriList[0] !== expectedfileSequences) {
-          //     error = `[${currTime}] Error in playlist! (BW:${bw}) Faulty Segment Continuity! Expected first item-uri in mseq(${variant.get(
-          //       "mediaSequence"
-          //     )}) to be: '${expectedfileSequences}'. Got: '${currentSegUriList[0]}'`;
-          //     console.error(`[${baseUrl}]${error}`);
-          //     data.errors.push(error);
-          //   }
-          // }
+          // Validate media sequence and file sequence
+          const mseqDiff = variant.get("mediaSequence") - data.variants[bw].mediaSequence;
+          if (mseqDiff < data.variants[bw].fileSequences.length) {
+            const expectedfileSequence = data.variants[bw].fileSequences[mseqDiff];
+
+            // if (currentSegUriList[0] !== expectedfileSequence) {
+            //   error = `[${currTime}] Error in playlist! (BW:${bw}) Faulty Segment Continuity! Expected first item-uri in mseq(${variant.get(
+            //     "mediaSequence"
+            //   )}) to be: '${expectedfileSequence}'. Got: '${currentSegUriList[0]}'`;
+            //   console.error(`[${baseUrl}]${error}`);
+            //   data.errors.push(error);
+            // }
+
+            // [!] Compare the end of filename instead...
```

Comment:
I thought since this kind of validation is still valuable to detect, to customize it so that we only compare part of the URI string. Limitation! Now this assumes that the 2 chars closest to the file extension reflect the segment index in some way.
ei.

"https://my.vod.com/vod/profile/720p/seg_720p_14.ts?somequery=ok" -> "14.ts" <- compare these parts between two segments in a media playlist

Reply:
Hopefully stealth bandwidth changes won't trigger this validation any more